File: src/Javascript/WorkSpace.js
var treeFile = "./workspace.json"

function FileTree() {
    this.update = function(files) {
        $("#FileStructure").jstree({
            'core' : {
              'data' : files}
          });
    }

    $('#FileStructure').on("changed.jstree", function (e, data) {
        console.log("The selected nodes are:");
        console.log(data.selected);
      });
}

// [
//   { "text" : "Project1", "children" : [
//       { "text" : "Suite1" , "children":[
//           {"text":"case1"},
//           {"text":"case2"}
//       ]},
//       { "text" : "Suite2" , "children": [
//           {"text":"case2"}
//       ]}
//     ]}
// ]

File: VisualScript/src/Ctrl/WorkSpace.py
from browser import window, alert
from File.WorkSpace import WorkSpace as wsModel
import os

class WorkSpace:
    def __init__(self):
        self.UI = window.FileTree.new()
        path = 'D:'
        p = ['Project', {'Project':{'Suite1': ['case1', 'case2'],
                          'Suite2': ['case2']}}]
        self.space = wsModel(path, p)
        json = self.space.getTreeJSON()
        self.UI.update(json)
        # self.UI.right = self.selected

    def selected(self, e, d):
        print("The selected node is: ")
        print(d)
